File: GFG/JavaScript/DSA/Easy/first-repeating-element.js
// https://www.geeksforgeeks.org/problems/first-repeating-element4018/1

class Solution {
  // Function to return the position of the first repeating element.
  firstRepeated(arr) {
    // your code here
    let map = {};
    const n = arr.length;
    let res = n;
    for (let i = 0; i < n; i++) {
      if (arr[i] in map) {
        if (res > map[arr[i]]) res = map[arr[i]];
      } else {
        map[arr[i]] = i;
      }
    }
    return res === n ? -1 : res + 1;
  }
}

File: GFG/Python/DSA/Easy/first-repeating-element.py
class Solution:
    # Function to return the position of the first repeating element.
    def first_repeated(self, arr):

        # arr : given array
        # n : size of the array
        mapper = {}
        n = len(arr)
        res = n

        for i in range(0, n):
            if arr[i] in mapper:
                if res > mapper[arr[i]]:
                    res = mapper[arr[i]]
            else:
                mapper[arr[i]] = i
        return -1 if res == n else res+1
